src/pages/Home.js

```diff
@@ -16,11 +16,51 @@ export default function Home(props) {
     history.push('/list');
   }
 
+  function handleInputValue(e) {
+    setInputValue(e.target.value.trim());
+  }
+
+  function checkExistingToken(e) {
+    e.preventDefault();
+    if (!inputValue) {
+      return swal('Please enter your token!', 'Input is empty', 'error');
+    }
+    db.collection(inputValue)
+      .get()
+      .then((snap) => {
+        if (snap.empty) {
+          swal(
+            'Token not found!',
+            'Please try again or start a new list!',
+            'error',
+          );
+        } else {
+          addKeyValuePairToLocalStorage('token', inputValue);
+          props.setToken(inputValue);
+          history.push('/list');
+        }
+      });
+  }
+
   return (
     <main>
-      {retrievedToken ? history.push('/list') : ''}
       <h1>Welcome screen</h1>
+      <h2>Add a new list:</h2>
       <button onClick={handleClick}>Add List</button>
+      <h2>Join an existing list</h2>
+      <form onSubmit={checkExistingToken}>
+        <label htmlFor="token">
+          Please Insert Token:
+          <input
+            type="text"
+            name="token"
+            placeholder="Three word token"
+            value={inputValue}
+            onChange={handleInputValue}
+          />
+        </label>
+        <input type="submit" value="Submit" />
+      </form>
```

Comment:
Good use of semantic and accessible form code!

Reply:
Thank you Yenly!